### ISSUE TITLE
DirectoryInfo.Exists returns incorrect results when it is cast from FileSystemInfo

### ISSUE THREAD

**Issue by troydai** [dotnet/corefx#616]
When test the same directory, `DirectoryInfo` and `Directory.Exist(...)` gives different answers. The bug is repo on `System.IO.FileSystem 4.0.0-beta-22530`

Repo code:
```
var root = Path.GetTempFileName();
File.Delete(root);
Directory.CreateDirectory(root);

var testDir = new DirectoryInfo(root);
var betaDir = Path.Combine(root, "beta");

Directory.CreateDirectory(betaDir);

var beta = testDir.EnumerateFileSystemInfos("beta", SearchOption.TopDirectoryOnly).First() as DirectoryInfo;
Console.WriteLine("Test {0} using DirectoryInfo: {1}", beta.FullName, beta.Exists);
Console.WriteLine("Test {0} using Directory.Exist: {1}", betaDir, Directory.Exists(betaDir));
```

**Comment by troydai** [comment]
I guess I file this bug at wrong place. I'll move it to https://github.com/dotnet/coreclr.

**Comment by troydai** [comment]
Reopen, as suggested by crew from coreclr the issue should be indeed opened here.

**Comment by ericstj** [comment]
I see.  I missed initializing _asDirectory in WIN32_FIND_DATA constructor.

**Comment by stephentoub** [comment]
Fixed in #640
